template/fr.opensagres.xdocreport.template.velocity/src/main/java/fr/opensagres/xdocreport/template/velocity/cache/XDocReportEntryResourceLoader.java

```diff
@@ -76,7 +77,7 @@ public Reader getResourceReader(String source, String encoding) throws ResourceN
             InputStream inputStream = cacheInfo.getInputStream();
             if ( inputStream != null )
             {
-                return new InputStreamReader(inputStream);
+                return new InputStreamReader(inputStream, Charset.forName(encoding));
```

Comment:
What about if encoding is null?

Reply:
You are right. I didn't think that the encoding could be null.

Reply:
i dont master the velocity code but I think you should take care of encoding which is null or empty.

Reply:
Yes,  I watched velocity code. It's really possible that encoding will be null.
I fixed this and created another pull request.